src/Facets/CalldataVerificationFacet.sol

```diff
@@ -0,0 +1,240 @@
+                data[4:],
+                (ILiFi.BridgeData, LibSwap.SwapData[])
+            );
+            sendingAssetId = swapData[0].sendingAssetId;
+            minAmount = swapData[0].fromAmount;
+        } else {
+            sendingAssetId = bridgeData.sendingAssetId;
+            minAmount = bridgeData.minAmount;
+        }
+
+        return (
+            bridgeData.bridge,
+            sendingAssetId,
+            bridgeData.receiver,
+            minAmount,
+            bridgeData.destinationChainId,
+            bridgeData.hasSourceSwaps,
+            bridgeData.hasDestinationCall
+        );
+    }
+
+    /// @notice Validates the calldata
+    /// @param data The calldata to validate
+    /// @param bridge The bridge to validate or empty string to ignore
+    /// @param sendingAssetId The sending asset id to validate
+    ///        or 0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF to ignore
+    /// @param receiver The receiver to validate
+    ///        or 0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF to ignore
+    /// @param amount The amount to validate or type(uint256).max to ignore
+    /// @param destinationChainId The destination chain id to validate
+    ///        or type(uint256).max to ignore
+    /// @param hasSourceSwaps Whether the calldata has source swaps
+    /// @param hasDestinationCall Whether the calldata has a destination call
+    /// @return isValid Whether the calldata is validate
+    function validateCalldata(
+        bytes calldata data,
+        string calldata bridge,
+        address sendingAssetId,
+        address receiver,
+        uint256 amount,
+        uint256 destinationChainId,
+        bool hasSourceSwaps,
+        bool hasDestinationCall
+    ) external pure returns (bool isValid) {
+        ILiFi.BridgeData memory bridgeData;
+        (
+            bridgeData.bridge,
+            bridgeData.sendingAssetId,
+            bridgeData.receiver,
+            bridgeData.minAmount,
+            bridgeData.destinationChainId,
+            bridgeData.hasSourceSwaps,
+            bridgeData.hasDestinationCall
+        ) = extractMainParameters(data);
+        return
+            (keccak256(abi.encodePacked(bridge)) ==
+                keccak256(abi.encodePacked("")) ||
+                keccak256(abi.encodePacked(bridgeData.bridge)) ==
+                keccak256(abi.encodePacked(bridge))) &&
+            (sendingAssetId == 0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF ||
+                bridgeData.sendingAssetId == sendingAssetId) &&
+            (receiver == 0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF ||
+                bridgeData.receiver == receiver) &&
+            (amount == type(uint256).max || bridgeData.minAmount == amount) &&
+            (destinationChainId == type(uint256).max ||
+                bridgeData.destinationChainId == destinationChainId) &&
+            bridgeData.hasSourceSwaps == hasSourceSwaps &&
+            bridgeData.hasDestinationCall == hasDestinationCall;
```

Comment:
Splitting this in more lines with comments would help to understand what we check here.